PyInstaller/building/makespec.py

```diff
@@ -309,6 +309,14 @@ def __add_options(parser):
                          "process signals both the bootloader and child "
                          "(e.g. via a process group) to avoid signalling "
                          "the child twice."))
+    g.add_argument('--pyiboot01-preserved-envs',
+                   action='append', default=['VIRTUAL_ENV'],
+                   metavar="ENVVAR", dest='pyiboot01_preserved_envs',
+                   help='Environment variables that will be preserved in pyiboot01_bootstrap module, '
+                   'the preserved env var will be renamed with prefix `PYINSTALLER_PRESERVED_`.'
+                   'This option can be used multiple times. (default: VIRTUAL_ENV)')
```

Comment:
* Option should be just `--preserve-envvar`. We must not bother users with `pyiboot01`, which is an implementation detail. Also "pyiboot01_bootstrap module" should not occur in the help-message.

* Also this should be singular, since the option adds a single variable to the list.

* I'd not add a default here. Most applications do not even `VIRTUAL_ENV`. Also be can omit the `--no-…` option.

* "Environment variable" (singular!) "to be preserved when running the application".

* "preserve" might not be the proper verb at all. I'm not a native speaker, but "retain", "copy", "keep" might be better.

Reply:
I finally decided to call it `--disguise-envvar`, as a verb-object phrase, I found the reason why I want to use the word "preserve" is because I only thought about the result of the env vars, they are indeed preserved, but our purpose is to hide them, or disguise, which has the meaning of change things to a different form, fits better in this situation.

Here's the LDOCE thesaurus of the two words:

- hide: to make something difficult to see or find, or to not show your true feelings
- disguise: to make someone or something seem like a different person or thing, so that other people cannot recognize them

I'm not a native speaker either, I agree we should choose the word very cautiously, glad to change it to other word if we can find out anything better.

Reply:
> I'd not add a default here. Most applications do not even VIRTUAL_ENV. Also be can omit the --no-… option.

I actually don't want to add the default value either, but I thought it might be friendly to not change the default behavior while adding new features, if you think it's OK, then I'd be happy to remove it and the `--no...` option (they were already removed in the latest code!).